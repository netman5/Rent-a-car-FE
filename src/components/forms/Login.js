import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { logUserToApi } from '../../redux/reducers/userReducer';

const Login = () => {
  const [state, setState] = useState({});
  const { error } = useSelector((state) => state.user);
  const dispatch = useDispatch();

  const onchange = (e) => {
    setState({
      ...state,
      [e.target.name]: e.target.value,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(logUserToApi(state));
  };

  return (
    <div className="container-fluid">
      <div className="row">
        <div className="column mt-5">
          <div className="d-flex justify-content-center align-items-center flex-column shadow-lg rounded w-50 mx-auto p-5">
            <h3 className="text-center">SIGN IN </h3>
            <br />
            {error ? (
              <div>
                {typeof error === 'string'
                  ? (<span>{error}</span>)
                  : (error.map((errorItem) => (
                    <span key={errorItem}>{errorItem}</span>
                  ))
                  )}
              </div>
            ) : ''}
            <form>
              <div>
                <label htmlFor="email" className="form-label">
                  Email address
                  <input type="email" data-testid="email-input" placeholder="Enter email" onChange={onchange} id="email" name="email" className="form-control" />
                </label>
              </div>
              <br />
              <div>
                <label htmlFor="password" className="form-label">
                  Password
                  <input type="password" data-testid="password-input" placeholder="Password" onChange={onchange} id="password" name="password" className="form-control" />
                </label>
              </div>
              <br />
              <button type="submit" className="btn btn-secondary" onClick={handleSubmit}>
                sign in
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login;
