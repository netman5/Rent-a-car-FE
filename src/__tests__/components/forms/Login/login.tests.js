import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { BrowserRouter as Router } from 'react-router-dom';

import store from '../../../../redux/configureStore';

import Login from '../../../../components/forms/Login';

it('Login renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Login />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
