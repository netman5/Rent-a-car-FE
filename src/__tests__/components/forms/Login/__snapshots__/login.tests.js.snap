// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="column mt-5"
    >
      <div
        className="d-flex justify-content-center align-items-center flex-column shadow-lg rounded w-50 mx-auto p-5"
      >
        <h3
          className="text-center"
        >
          SIGN IN 
        </h3>
        <br />
        <form>
          <div>
            <label
              className="form-label"
              htmlFor="email"
            >
              Email address
              <input
                className="form-control"
                data-testid="email-input"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Enter email"
                type="email"
              />
            </label>
          </div>
          <br />
          <div>
            <label
              className="form-label"
              htmlFor="password"
            >
              Password
              <input
                className="form-control"
                data-testid="password-input"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />
            </label>
          </div>
          <br />
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type="submit"
          >
            sign in
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
